### ISSUE TITLE
Set `prefers-color-scheme` CSS media query depending on whether or not a dark theme is set.

### ISSUE THREAD

**Issue by Anomalocaridid** [atlas-engineer/nyxt#2161]
**Is your feature request related to a problem? Please describe.**
Having used browsers like Firefox for a long time prior to moving to Nyxt, I took for granted how some websites would detect if I was using a dark theme and use a dark color scheme to match. Now that I have moved to Nyxt, this does not happen and I need to either manually change websites' color schemes or bear with it if it is not an option.

**Describe the solution you'd like**
I would like Nyxt to set the value of the [`prefers-color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) CSS query based on the value of `dark-p` for the currently set theme.

**Describe alternatives you've considered**
I could manually set the dark theme for websites that support it, but not all websites that support a light and dark theme have the option to manually set it. Also, I have to re-set it whenever I clear Nyxt's cookies, which I find to be a bit inconvenient. I am unaware of any existing features in Nyxt that could address this.

**Additional context**
N/A

**Comment by efimerspan** [comment]
@aartaka Chiming in to say I'd want this too. I currently set my color scheme  preference via the variable `gtk-application-prefer-dark-theme` in `~/.config/gtk-3.0/settings.ini`, but I'd want this to be dynamic for the current Nyxt session, meaning if I change my browser color scheme with [nx-mapper](https://github.com/efimerspan/nx-mapper) on the fly, I'd like this setting to be applied too and not have to close the browser, change the configuration file, and reopen a Nyxt instance. I've also tried to play around with [XSettings](https://github.com/derat/xsettingsd) but, similarly, this relies on a configuration file, and other alternatives such as `xrdb` require you to restart the application for the changes to take effect. 

One of the proposed solutions I've seen is changing this preference per site via CSS via the `color-scheme` property as per [this article](https://webkit.org/blog/8840/dark-mode-support-in-webkit/), but I haven't had any luck getting it to work with a user style. Could this perhaps fit a `cl-webkit` setting? I'm not sure if WebKitGTK currently supports this setting but at least Android WebView, which is based on WebKit, does support it as evidenced by [this post](https://localcoder.org/letting-webview-on-android-work-with-prefers-color-scheme-dark).

**Comment by aartaka** [comment]
No, WebKitGTK cannot do that. It's pretty restrictive in lots of things :)

**Comment by Anomalocaridid** [comment]
Setting `gtk-application-prefer-dark-theme` in `~/.config/gtk-3.0/settings.ini` like @efimerspan did works for me.
